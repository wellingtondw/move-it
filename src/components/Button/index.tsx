const Button = () => (
  <button>
    React Avan√ßado
  </button>
)

export default Button