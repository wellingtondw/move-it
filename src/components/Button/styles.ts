import styled, { css } from 'styled-components'

export const Button = styled.button`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    background-color: ${theme.colors.blue};
    border: none;
    outline: none;
    cursor: pointer;

    &:focus-within {
      box-shadow: 0 0 0.5rem ${theme.colors.secondaryText};
    }

  `}
`

export const Icon = styled.div`
${({ theme }) => css`
    width: ${theme.spacings.large};
    height:  ${theme.spacings.large};

    & > svg {
      width: 100%;
      color: ${theme.colors.white};
    }
  `}
`