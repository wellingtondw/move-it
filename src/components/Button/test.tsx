import { render, screen } from '@testing-library/react'

import Button from '.'

describe('<Button />', () => {
  it('should render the button correctly', () => {
    const { container } = render(<Button />)

    expect(
      screen.getByRole('button', { name: /react avan√ßado/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})