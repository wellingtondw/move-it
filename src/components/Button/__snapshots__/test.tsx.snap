// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the button 1`] = `
<button>
  React Avançado
</button>
`;

exports[`<Button /> should render the button correctly 1`] = `
<button>
  React Avançado
</button>
`;
